File: practicas/prt42/circulo.py
from math import pi

class Figura:
    def __init__(self, color):
        self.color = color

    def area(self):
        raise NotImplementedError

    def perimetro(self):
        raise NotImplementedError

    def __str__(self):
        return f"Color: {self.color}"

class Circulo(Figura):
    def __init__(self, radio, color):
        super().__init__(color)
        self.radio = radio

    def area(self):
        return pi * self.radio ** 2

    def perimetro(self):
        return 2 * pi * self.radio

    def __str__(self):
        return f"Circulo - {super().__str__()}, Radio: {self.radio}"

File: practicas/prt42/circulo.java
package practicas.prt42;

public class circulo extends figura {
    private double radio;

    public circulo(double radio, String color) {
        super(color);
        this.radio = radio;
    }

    @Override
    public double area() {
        return Math.PI * radio * radio;
    }

    @Override
    public double perimetro() {
        return 2 * Math.PI * radio;
    }

    @Override
    public String toString() {
        return "Circulo - " + super.toString() + ", Radio: " + radio;
    }
}
